src/current_user.ts

```diff
@@ -39,10 +40,40 @@ export interface CurrentUserOptions {
   avatarURL?: string;
   customData?: any; // TODO: Shouldn't be any (type)
   rooms?: Room[];
-  instance: Instance;
+  apiInstance: Instance;
+  filesInstance: Instance;
   userStore: GlobalUserStore;
 }
 
+export interface DataAttachment {
+  file: Blob;
+  name: string;
+}
+
+export interface LinkAttachment {
+  link: string;
+  type: string;
```

Comment:
Are `LinkAttachments`s all of the type "link", or are types arbitrary?

If types belong to some fixed set, it might be nice to put them in an enum?

Reply:
I'd like to enum-ify the `type` property eventually but they're still in flux and so for now I think the tradeoff of allowing some typo mistakes is worth it for the flexibility.